tests/pytest/eligibility/test_views.py

```diff
@@ -52,3 +67,201 @@ def test_start_without_verifier(mocker, client):
+    body = {"sub": "A7654321", "name": "Garcia"}
+    request = rf.post(path, body)
+
+    initialize_request(request)
+    session.update(request, agency=agency, verifier=verifier, oauth_token="token")
+
+    with pytest.raises(ApiError):
+        confirm(request)
+
+
+@httpretty.activate(verbose=True, allow_net_connect=False)
+@pytest.mark.django_db
+def test_confirm_failure_unexpected_status_code(mocker, rf):
+    agency, verifier = set_verifier(mocker)
+
+    httpretty.register_uri(httpretty.GET, "http://localhost/verify", status=404)
+
+    path = reverse("eligibility:confirm")
+    body = {"sub": "A1234567", "name": "Garcia"}
+    request = rf.post(path, body)
+
+    initialize_request(request)
+    session.update(request, agency=agency, verifier=verifier, oauth_token="token")
+
+    with pytest.raises(ApiError, match=r"Unexpected eligibility"):
+        confirm(request)
+
+
+@httpretty.activate(verbose=True, allow_net_connect=False)
+@pytest.mark.django_db
+def test_confirm_failure_error_tokenizing_request(mocker, rf):
+    agency, verifier = set_verifier(mocker)
+    agency.jws_signing_alg = "not real"
+
+    path = reverse("eligibility:confirm")
+    body = {"sub": "A0101011", "name": "Lastname"}
+    request = rf.post(path, body)
+
+    initialize_request(request)
+    session.update(request, agency=agency, verifier=verifier, oauth_token="token")
+
+    with pytest.raises(TokenError):
+        confirm(request)
+
+
+def _tokenize_response_error_scenarios():
+    return [
+        pytest.param(lambda verifier: "", id='TokenError("Invalid response format")'),
+        pytest.param(lambda verifier: "invalid token", id='TokenError("Invalid JWE token")'),
+        # Can't figure out the right way to cause these cases to be thrown.
+        # pytest.param(lambda verifier: _make_token(
+        #     {
+        #         "jti": str(uuid.uuid4()),
+        #         "iss": "test-server",
+        #         "iat": int(datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).timestamp()),
+        #         "eligibility": ["type1"],
+        #     },
+        #     verifier.jws_signing_alg,
+        #     verifier.jws_signing_alg,
+        #     _get_jwk("server.key"),
+        #     verifier.jwe_encryption_alg,
+        #     verifier.jwe_cek_enc,
+        #     _get_jwk("client.pub"),
+        # ), id='TokenError("JWE token decryption failed")'),
+        # pytest.param(lambda verifier: _make_token(
+        #     {
+        #         "jti": str(uuid.uuid4()),
+        #         "iss": "test-server",
+        #         "iat": int(datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).timestamp()),
+        #         "eligibility": ["type1"],
+        #     },
+        #     verifier.jws_signing_alg,
+        #     _get_jwk("server.key"),
+        #     verifier.jwe_encryption_alg,
+        #     verifier.jwe_cek_enc,
+        #     _get_jwk("client.pub"),
+        # ), id='TokenError("Invalid JWS token")'),
```

Comment:
Can we put this in an issue instead? Avoids clutter in the codebase and makes it more clearly a TODO.

Reply:
Created #585